atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -1566,6 +1576,12 @@ private void createTablesInternal(final Map<TableReference, byte[]> tableNamesTo
         } else {
             log.debug("Found table metadata for {} at matching name {}", LoggingArgs.tableRef(tableRef),
                     LoggingArgs.tableRef("matchingTable", match.get().getKey()));
+
+            // rewrite this row with the key we initially expected, so we don't have to do this again
+            putMetadataAndMaybeAlterTables(
```

Comment:
Let's delete the old versions, since a future delete would clean up the lower case version and the stale versions would become visible again.